test/e2e/common.go

```diff
@@ -0,0 +1,370 @@
+			}
+
+			csr.Status.Conditions = append(csr.Status.Conditions, certificatesv1beta1.CertificateSigningRequestCondition{
+				Type:    certificatesv1beta1.CertificateApproved,
+				Reason:  "Approved by E2E",
+				Message: "Approved as part of e2e",
+			})
+			_, err := csrClient.UpdateApproval(context.TODO(), csr, metav1.UpdateOptions{})
+			return err
+		})
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
+
+func isCSRInTerminalState(status *certificatesv1beta1.CertificateSigningRequestStatus) bool {
+	for _, c := range status.Conditions {
+		if c.Type == certificatesv1beta1.CertificateApproved {
+			return true
+		}
+		if c.Type == certificatesv1beta1.CertificateDenied {
+			return true
+		}
+	}
+	return false
+}
+
+func acceptsClient(clusterName string) {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var err error
+		managedCluster, err := clusterClient.ClusterV1().ManagedClusters().Get(context.TODO(),
+			clusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		managedCluster.Spec.HubAcceptsClient = true
+		managedCluster.Spec.LeaseDurationSeconds = 5
+		managedCluster, err = clusterClient.ClusterV1().ManagedClusters().Update(context.TODO(),
+			managedCluster, metav1.UpdateOptions{})
+		return err
+	})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func managedClusterReady(clusterName string) error {
+	managedCluster, err := clusterClient.ClusterV1().ManagedClusters().Get(context.TODO(),
+		clusterName, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	var okCount = 0
+	for _, condition := range managedCluster.Status.Conditions {
+		if (condition.Type == clusterv1.ManagedClusterConditionHubAccepted ||
+			condition.Type == clusterv1.ManagedClusterConditionJoined ||
+			condition.Type == clusterv1.ManagedClusterConditionAvailable) &&
+			condition.Status == v1beta1.ConditionTrue {
+			okCount++
+		}
+	}
+
+	if okCount == 3 {
+		return nil
+	}
+
+	return fmt.Errorf("condtions are not ready: %v", managedCluster.Status.Conditions)
+}
+
+func newConfigmap(namespace, name string, data map[string]string) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Data: data,
+	}
+}
+
+func createWorkOfConfigMap(name, clusterName, configMapName, configMapNamespace string) {
+	manifest := workapiv1.Manifest{}
+	manifest.Object = newConfigmap(configMapNamespace, configMapName, map[string]string{"a": "b"})
+	manifestWork := &workapiv1.ManifestWork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: workapiv1.ManifestWorkSpec{
+			Workload: workapiv1.ManifestsTemplate{
+				Manifests: []workapiv1.Manifest{
+					manifest,
+				},
+			},
+		},
+	}
+	_, err := workClient.WorkV1().ManifestWorks(clusterName).
+		Create(context.TODO(), manifestWork, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func cleanResources() {
```

Comment:
Be careful about this - we should be able to run the e2e against a live production system as part of a health check if we want to. As written, this will wipe a system :)

Reply:
refactor to only delete resources created by e2e tests